cmd/lambda/accounts/list.go

```diff
@@ -16,24 +19,79 @@ import (
 // GetAllAccounts - Returns all the accounts.
 func GetAllAccounts(w http.ResponseWriter, r *http.Request) {
 	// Fetch the accounts.
-	accounts, err := Dao.GetAccounts()
+
+	getAccountsInput, err := parseGetAccountsInput(r)
 
 	if err != nil {
-		errorMessage := fmt.Sprintf("Failed to query database: %s", err)
-		log.Print(errorMessage)
-		api.WriteAPIErrorResponse(w, errors.NewInternalServer(errorMessage, nil))
+		response.WriteRequestValidationError(w, fmt.Sprintf("Error parsing query params: %s", err))
+		return
+	}
+
+	result, err := Dao.GetAccounts(getAccountsInput)
+
+	if err != nil {
+		response.WriteServerErrorWithResponse(w, fmt.Sprintf("Error querying accounts database: %s", err))
```

Comment:
Can we log the `err` instead of returning it? Internal server errors may contain private info that we do not want to expose to the client.

Reply:
Yes, fixed it for both leases/list.go and accounts/list.go